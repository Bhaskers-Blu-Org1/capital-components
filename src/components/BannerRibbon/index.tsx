import Ribbon from "./BannerRibbon";
import Breadcrumb from "./Breadcrumb";
import DropdownWrapper from "./DropdownWrapper";

export default {
  Ribbon,
  DropdownWrapper,
  Breadcrumb
};
