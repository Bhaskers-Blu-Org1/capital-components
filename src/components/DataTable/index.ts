export { default, IColumnDescriptor } from "./DataTable";
export * from "./utils";
