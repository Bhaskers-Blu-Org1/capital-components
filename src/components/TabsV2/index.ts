export { default } from "./TabsV2";
