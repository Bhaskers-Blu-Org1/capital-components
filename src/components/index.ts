export { default as Col } from "./Col";
export { default as Grid } from "./Grid";
export { default as DataTable } from "./DataTable";
export { default as FlyOver } from "./FlyOver";
export { default as FlyOverProvider } from "./FlyOverProvider";
export { default as Icon } from "./Icon";
export { default as Table } from "./Table";
export { default as Popover } from "./Popover";
export { default as DropMenu } from "./DropMenu";
export { default as DropMenuItem } from "./DropMenuItem";
export { default as Tabs } from "./Tabs";
export { default as BleedingRow } from "./BleedingRow";
export { default as PushOver } from "./PushOver";
export { default as PushOverItem } from "./PushOverItem";
export { default as TabsV2 } from "./TabsV2";
export { default as Tab } from "./Tab";
