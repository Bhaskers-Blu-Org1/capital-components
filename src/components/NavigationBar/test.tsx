import { cleanup, getByLabelText as getByLabelTextGlobal } from "react-testing-library";
import "jest-dom/extend-expect";
import { NavigationBar } from "@fss/components";
import React from "react";
import { renderWithDefaultTheme, mockScreenSize } from "test-utils";

const { PrimaryBar, PrimaryBarIcon, PrimaryBarNavItem, PrimaryBarTitle } = NavigationBar;

afterEach(cleanup);

test("Basic snapshot, in monitor", async () => {
  mockScreenSize("monitor");
  const { baseElement } = renderWithDefaultTheme(
    <PrimaryBar
      titleSection={<PrimaryBarTitle>Title</PrimaryBarTitle>}
      navSection={
        <>
          <PrimaryBarNavItem>Item 1</PrimaryBarNavItem>
          <PrimaryBarNavItem>Item 2</PrimaryBarNavItem>
        </>
      }
      rightSection={<PrimaryBarIcon>ICO</PrimaryBarIcon>}
    />
  );

  expect(baseElement).toMatchSnapshot();
});

test("Open menu, in phone", async () => {
  mockScreenSize("phone");
  const { getByRole } = renderWithDefaultTheme(
    <PrimaryBar
      id="main"
      mobileMenuRef={document.body}
      showMenu={true}
      titleSection={<PrimaryBarTitle>Title</PrimaryBarTitle>}
      navSection={
        <>
          <PrimaryBarNavItem>Item 1</PrimaryBarNavItem>
          <PrimaryBarNavItem isSelected>Item 2</PrimaryBarNavItem>
        </>
      }
      rightSection={
        <>
          <PrimaryBarIcon>ICO</PrimaryBarIcon>
          <PrimaryBarIcon isSelected>ICO2</PrimaryBarIcon>
        </>
      }
      mobileMenuHeaderContent="Some content"
    />
  );

  const mainMenuButton = getByRole("button");
  expect(mainMenuButton).toBeInTheDocument();
  expect(mainMenuButton).toHaveAttribute(
    "aria-controls",
    "wfss-navigation-bar-primary-main-mobile-menu"
  );
  expect(getByLabelTextGlobal(document.body, "Menu")).toBeInTheDocument();
});
