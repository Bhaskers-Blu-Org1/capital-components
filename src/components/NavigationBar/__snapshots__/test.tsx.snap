// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Secondary Bar in phone 1`] = `
<body>
  .emotion-1 {
  font-weight: 400;
  overflow: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-bottom: #767676 1px solid;
  background-color: #252525;
}

@media (min-width:768px) {
  .emotion-1 {
    direction: rtl;
  }
}

@media (min-width:768px) {
  .emotion-1 [role="tablist"]::after,
  .emotion-1 [role="tablist"]::before {
    content: "";
    padding-left: 28px;
  }
}

@media (max-width:767px) {
  .emotion-1 [role="tablist"]::after,
  .emotion-1 [role="tablist"]::before {
    content: "";
    padding-left: 0.75rem;
  }

  .emotion-1 [role="tablist"] {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

<div>
    <nav
      class="emotion-0 emotion-1 emotion-2"
    >
      Item1 Item2
    </nav>
  </div>
</body>
`;

exports[`Basic snapshot, in monitor 1`] = `
.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 1px solid #97c1ff;
  background-color: #000000;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 48px;
  overflow: hidden;
}

@media (min-width:768px) {
  .emotion-12 {
    padding-left: 40px;
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.75rem 1rem;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 400;
  color: #ffffff;
  fill: #ffffff;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding-left: 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  cursor: auto;
  font-size: 1rem;
  line-height: 1.25rem;
  font-weight: 400;
  font-weight: 600;
}

.emotion-0:hover {
  background-color: #252525;
}

.emotion-0:hover {
  background-color: #000000;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: auto;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.75rem 1rem;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 400;
  color: #ffffff;
  fill: #ffffff;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-2:hover {
  background-color: #252525;
}

.emotion-10 {
  margin-left: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-10 > * {
  border-left: 1px solid #767676;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.75rem 1rem;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 400;
  color: #ffffff;
  fill: #ffffff;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  color: #ffffff;
  fill: #ffffff;
}

.emotion-8:hover {
  background-color: #252525;
}

.emotion-8:hover > .icon-circle {
  color: #252525;
  fill: #252525;
}

<body>
  <div>
    <nav
      aria-label="Main"
      class="emotion-12 emotion-13"
      id="wfss-navigation-bar-primary-"
    >
      <div
        class="emotion-0 emotion-1"
      >
        Title
      </div>
      <div
        class="emotion-6 emotion-7"
      >
        <div
          class="emotion-2 emotion-3"
        >
          Item 1
        </div>
        <div
          class="emotion-2 emotion-3"
        >
          Item 2
        </div>
      </div>
      <div
        class="emotion-10 emotion-11"
      >
        <div
          class="emotion-8 emotion-9"
        >
          ICO
        </div>
      </div>
    </nav>
  </div>
</body>
`;
